core/cli/commands.py

```diff
@@ -380,3 +382,14 @@ def validate(self, document):
                 raise ValidationError(
                     message='Enter a valid module name',
                     cursor_position=len(document.text))
+            elif document.text.lower() in [module.lower() for module in self.modules]:
+                raise ValidationError(
+                    message="Try a different name, this module name is available",
+                    cursor_position=len(document.text))
+
+class NamespaceValidation(Validator):
+    def validate(self, document):
+        if document.text == ("kernel" if check_os() else "firewall"):
```

Comment:
It doesn't seem convenient to me. We have to add bunch of new controls as we add new namespaces. So it should be replaced with something more appropriate for the context.

Reply:
Replaced with a more appropriate structure.